src/compute-client/src/controller/instance.rs

```diff
@@ -999,18 +999,17 @@ where
 
         // Validate sources have `since.less_equal(as_of)`.
         for source_id in dataflow.source_imports.keys() {
-            let since = &self
-                .storage_controller
-                .collection(*source_id)
-                .map_err(|_| DataflowCreationError::CollectionMissing(*source_id))?
-                .read_capabilities
-                .frontier();
-            if !(timely::order::PartialOrder::less_equal(since, &as_of.borrow())) {
-                Err(DataflowCreationError::SinceViolation(*source_id))?;
-            }
+            // We cannot assert on the _current_ implied capability/since of
+            // depended-upon storage collections. Their frontiers might move
+            // along concurrently. Our only hope is that our callers put in
+            // place the right read holds, which prevents sinces from advancing.
+            //
+            // Below, where we install read holds using
+            // update_read_capabilities, that call will fail if someone messed
+            // up the read holds.
```

Comment:
By "fail" you mean "panic", right? I guess there is nothing better to do here on short notice, but could you add a `TODO` saying that we should force the caller to pass in read holds and do the validation based on those? You can make it a `TODO(#24266)`.